### PR TITLE
feat(Controller): add Universal Controller support

### PR DESCRIPTION
**Pull request by christianalfoni**
Read the added docs to see how it works: [docs](https://github.com/cerebral/cerebral/blob/5ff7c0ed5ceccfa7236967bafd179f7ed86eeeea/docs/api/universalcontroller.md)

Not sure how the new testing works sp please give pointers for me to update that, getting dependency errors.

I suggest we replace the StateContainer with this.

### CODE REVIEW

File: demos/universal/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "start:server": "babel-node server --presets es2015,react"
+  },
+  "devDependencies": {
+    "babel-cli": "^6.24.1",
+    "concurrently": "^3.4.0",
+    "express": "^4.15.3",
+    "prop-types": "^15.5.10",
+    "webpack": "^2.5.1",
+    "webpack-dev-server": "^2.4.5"
+  }
```
Comment (henri-hulski) [inline demos/universal/package.json]: You should run in root directory of monorepo
```sh
npm install --save-dev --save-exact babel-cli@6.24.1 concurrently@3.4.0 express@4.15.3 prop-types@15.5.10 webpack@2.5.1 webpack-dev-server@2.4.5 mock-socket@6.0.4
```
To add the dependencies also to the monorepo.

@gaspard plans to automate this with a script.

File: docs/api/universalcontroller.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        res.send(index)
+      })
+})
+
+server.listen(3000)
+```
+
+## ES6 on server
+Take a look at the [demo application](https://github.com/cerebral/cerebral/tree/master/demos/universal) to see how you can run modern javascript, also with JSX, on the server. Demo does not include building for production, but you would typically use Webpack as normal for the client and just babel for the server. Or you can use Webpack there as well.
```
Comment (edgesoft) [inline docs/api/universalcontroller.md]: Link is dead

File: packages/cerebral/src/UniversalController.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/* eslint-env mocha */
+import UniversalController from './UniversalController'
+import assert from 'assert'
+
+describe.only('UniversalController', () => {
```
Comment (gaspard) [inline packages/cerebral/src/UniversalController.test.js]: Please remove 'only', coverage is not happy, half the tests are off :-)

**Comment by edgesoft** [comment]
Tested this in my project and works. This can be merged in.

**Comment by Guria** [comment]
conflicts after merging router

**Comment by Guria** [comment]
still conflicting
cc @gaspard

**Comment by edgesoft** [comment]
@Guria This is resolved

**Pull request merged**